docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,157 @@
+
+At a high level, this solution would require us to:
+ - Fix the build so it's not hardcoded to look for Intel MKL or any native binaries.
+ - Fix the managed code so that if it can't find the native binaries, the code behaves correctly and performs the software fallback.
+ - CpuMathNative has software fallbacks already in place for .NET core 3.1, so no work is needed.
+ - FastTreeNative has a flag for a software fallback. We would need to conditionally enable this. Alternatively, we could change the C# code so the software fallback is always enabled for the cases when it can't find the native binaries.
+ - LdaNative would need to be re-written.
+ - MatrixFactorizationNative would need to be re-written.
+ - MklProxyNative can be ignored. It is not needed with software fallbacks. We will need to modify the build to exclude it as needed.
+ - SymSgdNative would need to be re-written.
+
+### 3.4 Hybrid Between Finding Replacements and Software Fallbacks
+Since some of the native code already has replacements and some of the code already has software fallbacks, we can leverage this work by doing a hybrid between the prior 2 solutions.
+
+At a high level, this solution would require us to:
+ - Fix the build so it's not hardcoded to look for Intel MKL or any native binaries.
+ - Fix the managed code so that if it can't find Intel MKL or any native binaries, the code behaves correctly. This includes software fallbacks and/or description error messages.
+ - CpuMathNative has software fallbacks already in place for .NET core 3.1, so no work is needed.
+ - FastTreeNative has a flag for a software fallback. We would need to conditionally enable this. Alternatively, we could change the C# code so the software fallback is always enabled for the cases when it can't find the native binaries.
+ - LdaNative builds just fine for Arm, so no work would be required.
+ - MatrixFactorizationNative builds with the "USESSE" flag that requires Intel MKL. We can conditionally enable/disable that flag and no other work would be required.
+ - MklProxyNative can be removed/ignored for Arm builds. We will need to modify the build to exclude it as needed.
+ - SymSgdNative needs to be either re-written in managed code, or re-write 4 Intel MKL methods. The 4 methods are just dealing with vector manipulation and shouldn't be hard to do.
+
+## My Suggestion
+My suggestion would be to start with the hybrid approach. It will require the least amount of work to get ML.NET running elsewhere, while still being able to support a large majority of devices out of the gate. This solution will still limit the platforms we can run-on to Arm64 devices, but we can do a generic Arm64 compile, so it should work for all Arm64 v8 devices. It would be a good idea to have full software fallbacks so that truly anywhere .NET Core runs, ML.NET will run, such as Web Assembly and when .NET 6 comes out on mobile as well.
```

Comment:
Once a fallback is in place, I might recommend benchmarking the speed gain MKL gives us. 

If the speed gain is not significant, removing MLK will give the benefit of simplicity. Historically, there has been many filed [issues related to MLK](https://github.com/dotnet/machinelearning/search?q=MKL&type=issues) (build issues, runtime issues, maintenance issues).